src/density-qvm.lisp

```diff
@@ -123,10 +125,12 @@ POVM must be a 4-element list of double-floats."))
   (assert (typep (quil:application-operator instr) 'quil:named-operator) ; TODO XXX support gate modifiers
           (instr)
           "The density QVM doesn't support gate modifiers.")
-  (let*  ((gate (pull-teeth-to-get-a-gate instr))
+  (let*  ((gate-name (quil::application-operator-root-name instr))
           (params (mapcar (lambda (p) (force-parameter p qvm))
                           (quil:application-parameters instr)))
-          (qubits (mapcar #'quil:qubit-index (quil:application-arguments instr))))
+          (qubits (mapcar #'quil:qubit-index (quil:application-arguments instr)))
+          (sops (gethash (list gate-name qubits) (noisy-gate-definitions qvm)))
+          (gate (or sops (pull-teeth-to-get-a-gate instr))))
```

Comment:
Is this correct? Are the superoperators meant to completely replace the gate application (as opposed to being applied "on top of" the standard gate?  This appears to be analogous to how things are done over in `noisy-qvm`.

Reply:
yes they are